libs/feature/search/src/lib/state/search.facade.ts

```diff
@@ -87,39 +88,53 @@ export class SearchFacade {
     private filterGeometry: Promise<Geometry>
   ) {}
 
+  private searchExists(searchId: string): boolean {
+    let exists = false
+    this.store
+      .pipe(select(getSearchStateSearch, searchId))
+      .subscribe((search) => {
+        exists = !!search
+      })
+      .unsubscribe()
+    return exists
+  }
+
   init(searchId: string = DEFAULT_SEARCH_KEY): void {
-    if (this.searchId)
-      throw new Error(
-        `This SearchFacade instance was already initialized with the following searchId: ${this.searchId}`
+    if (searchId !== this.searchId) {
+      this.searchId = searchId
+      if (!this.searchExists(searchId)) {
+        this.store.dispatch(new AddSearch(searchId))
+      }
+
+      this.results$ = this.store.pipe(select(getSearchResults, searchId))
+      this.layout$ = this.store.pipe(select(getSearchResultsLayout, searchId))
+      this.isLoading$ = this.store.pipe(
+        select(getSearchResultsLoading, searchId)
```

Comment:
Hum, I don't really like the changes on the facade and don't think this should be necessary. The facades are already initialized via the `SearchRouterContainerDirective` in the `AppComponent` and the `SearchStateContainerDirective` in the `MyRecordsComponent` and a distinct instance should be available for each state.